### PR TITLE
Feature/u2f

### PR DESCRIPTION
**Pull request by fillup**

### CODE REVIEW

File: models/u2f.js [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+  console.log('Starting create U2F authentication for uuid: ' + u2fUuid);
+  apiKey.getActivatedApiKey(apiKeyValue, apiSecret, (error, apiKeyRecord) => {
+    if (error) {
+      console.log('Unable to get activated API Key in order to create U2F:', error);
+      response.returnError(401, 'Unauthorized', callback);
+      return;
+    }
+
+    if (!u2fUuid) {
+      console.log('U2F delete request had no UUID.');
```
Comment (Baggerone) [inline models/u2f.js]: Is "delete" right here?

File: models/u2f.js [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+      }
+
+      const encryptedKeyHandle = apiKeyRecord.u2f[u2fUuid].encryptedKeyHandle;
+      if (!encryptedKeyHandle) {
+        console.error('No encryptedKeyHandle found in that U2F record.');
+        response.returnError(500, 'Internal Server Error', callback);
+        return;
+      }
+
+      encryption.decrypt(encryptedKeyHandle, apiSecret, (error, keyHandle) => {
```
Comment (Baggerone) [inline models/u2f.js]: Can't we avoid nested multi-line anonymous callbacks?

File: models/u2f.js (hunk 1 above)
Comment (forevermatt) [reply models/u2f.js]: Good catch. I'll fix that.

Comment (forevermatt) [reply models/u2f.js]: This is kind of strange. This comment used to be on a different line (19), where "delete" wasn't the right word. Now it's on line 137, where "delete" is the right word. Oh well, the code is now correct in that regard.

**Pull request merged**